deps/rabbit/src/rabbit_stream_coordinator.erl

```diff
@@ -398,19 +441,63 @@ apply(Meta, {down, Pid, Reason} = Cmd,
         error ->
             return(Meta, State, ok, Effects0)
     end;
-apply(Meta, {register_listener, #{pid := Pid,
-                                  stream_id := StreamId}},
+apply(#{machine_version := MachineVersion} = Meta,
+      {register_listener, #{pid := Pid,
+                            stream_id := StreamId} = Args},
       #?MODULE{streams = Streams,
-               monitors = Monitors0} = State0) ->
-    case Streams of
-        #{StreamId := #stream{listeners = Listeners0} = Stream0} ->
+               monitors = Monitors0} = State0) when MachineVersion =< 1 ->
+    Type = maps:get(type, Args, leader),
+    case {Streams, Type} of
+        {#{StreamId := #stream{listeners = Listeners0} = Stream0}, leader} ->
             Stream1 = Stream0#stream{listeners = maps:put(Pid, undefined, Listeners0)},
-            {Stream, Effects} = eval_listeners(Stream1, []),
+            {Stream, Effects} = eval_listeners(MachineVersion, Stream1, []),
             Monitors = maps:put(Pid, {StreamId, listener}, Monitors0),
             return(Meta,
                    State0#?MODULE{streams = maps:put(StreamId, Stream, Streams),
                                   monitors = Monitors}, ok,
                    [{monitor, process, Pid} | Effects]);
+        {#{StreamId := _Stream}, local_member} ->
+            %% registering a local member listener does not change the state in v1
+            return(Meta, State0, ok, []);
+        _ ->
+            return(Meta, State0, stream_not_found, [])
+    end;
+
+apply(#{machine_version := MachineVersion} = Meta,
+      {register_listener, #{pid := Pid,
+                            stream_id := StreamId} = Args},
+      #?MODULE{streams = Streams,
+               monitors = Monitors0} = State0) when MachineVersion >= 2 ->
+    Node = case Args of
```

Comment:
Might be better to use `maps:get/3` here and for `type` below to provide the defaults.